### PR TITLE
vec2text evaluation

### PR DESCRIPTION
**Pull request by basiekjusz**
This PR contains implementation of vec2text model and it's evaluation methods.

Please note that vec2text uses gtr-t5-base embedding model.

### CODE REVIEW

File: tml_decoder/models/vec2text_model.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        self.num_steps = num_steps
+        self.device = device
+
+    @property
+    def name(self):
+        return "vec2text model"
+
+    def get_label(self, texts: Iterable[str]) -> str:
+        embeddings = self.encoder.encode(texts, convert_to_tensor=True).to(self.device)
+        mean_embedding = embeddings.mean(dim=0, keepdim=True)
```
Comment (dmkwis) [inline tml_decoder/models/vec2text_model.py]: Not like this:
embeddings.sum(dim=0)
and then normalize the result to have norm one

**Review by dmkwis — APPROVED** [review]
One thingy to fix

File: tml_decoder/models/vec2text_model.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+from vec2text import invert_embeddings, load_corrector, models
+from sentence_transformers import SentenceTransformer
+import torch
+
+
+class Vec2TextModel(AbstractLabelModel):
+    def __init__(self, num_steps=10, device="mps"):
+        self.corrector = load_corrector("gtr-base")
+        self.encoder = SentenceTransformer(
+            "sentence-transformers/gtr-t5-base", device=device
```
Comment (dmkwis) [inline tml_decoder/models/vec2text_model.py]: Are we sure that the weights of this model are the same ones that the vec2text was trained on?

File: tml_decoder/models/vec2text_model.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+from tml_decoder.models.abstract_model import AbstractLabelModel
+from vec2text import invert_embeddings, load_corrector, models
+from sentence_transformers import SentenceTransformer
+import torch
+
+
+class Vec2TextModel(AbstractLabelModel):
+    def __init__(self, num_steps=10, device="mps"):
+        self.corrector = load_corrector("gtr-base")
+        self.encoder = SentenceTransformer(
```
Comment (dmkwis) [inline tml_decoder/models/vec2text_model.py]: Maybe also make class inheriting from abstract_encoder wrapping this model

**Review by dmkwis — COMMENTED** [review]
Also made some comments about wrapping the encoder

File: tml_decoder/models/vec2text_model.py (hunk 2 above)
Comment (basiekjusz) [reply tml_decoder/models/vec2text_model.py]: Yes, `vec2text` supports only two encoders and `gtr-base` is one of them. It's in their docs as well.

File: tml_decoder/encoders/mini_lm_encoder.py [hunk 4]
```diff
@@ -11,15 +11,18 @@ def __init__(self):
...
+        result = self.encoder.encode(text)
+        if isinstance(result, np.ndarray):
+            return result
+        
+        raise TypeError("Expected ndarray from self.encoder.encode, got {}".format(type(result)))
     
     @property
     def name(self):
-        return "Mini LM"
+        return "gtr-base"
```
Comment (dmkwis) [inline tml_decoder/encoders/mini_lm_encoder.py]: to nie jest gtr-base

File: tml_decoder/utils/common_utils.py [hunk 5]
```diff
@@ -82,5 +86,7 @@ def get_model(name: str, *args: Any, **kwargs: Any) -> AbstractLabelModel:
     if name == "MCTS":
         generator = get_generator("gpt2")
         return MCTSModel(generator=generator, *args, **kwargs)
```
Comment (dmkwis) [inline tml_decoder/utils/common_utils.py]: we also need to pass an encoder here

**Review by dmkwis — COMMENTED** [review]
2 fixes

File: tml_decoder/encoders/mini_lm_encoder.py (hunk 4 above)
Comment (basiekjusz) [reply tml_decoder/encoders/mini_lm_encoder.py]: xDD

Comment (basiekjusz) [reply tml_decoder/encoders/mini_lm_encoder.py]: przepraszam coś mi się kliknęło i nie wiem jak to cofnąć
